ext/mysql2/extconf.rb

```diff
@@ -159,6 +159,7 @@ def add_ssl_defines(header)
 have_const('SERVER_QUERY_WAS_SLOW', mysql_h)
 have_const('MYSQL_OPTION_MULTI_STATEMENTS_ON', mysql_h)
 have_const('MYSQL_OPTION_MULTI_STATEMENTS_OFF', mysql_h)
+have_const('MYSQL_OPT_GET_SERVER_PUBLIC_KEY', mysql_h)
```

Comment:
It's not necessary to test for this constant into its own HAVE_ constant, because we can directly test for the constant at time of use (see suggested change)

Reply:
@sodabrew Hello, Thank you for reviewing my PR.
`MYSQL_OPT_GET_SERVER_PUBLIC_KEY` is defined as `enum`. If my understand correctly, `ifdef` does not work with `enum`.
https://github.com/mysql/mysql-server/blob/8.0/include/mysql.h.pp#L411-L448

Reply:
You’re absolutely right!